server/src/main/java/io/crate/execution/engine/sort/Sort.java

```diff
@@ -44,26 +52,42 @@ public static <T> CompletableFuture<List<T>> parallelSort(List<T> list,
         int itemsPerThread = list.size() / Math.max(1, numAvailableThreads);
         if (numAvailableThreads <= 1 || itemsPerThread < minItemsPerThread) {
             list.sort(comparator);
+            allocateBytes.accept(sortOverheadBytes(list.size()));
             return completedFuture(list);
         }
         List<List<T>> partitions = Lists.partition(list, itemsPerThread);
-        ArrayList<CompletableFuture<List<T>>> futures = new ArrayList<>(partitions.size());
+        allocateBytes.accept(RamUsageEstimator.sizeOfObject(partitions)); // Accounts for nested list
+        CompletableFuture<List<T>>[] futures = new CompletableFuture[partitions.size()];
+        allocateBytes.accept(RamUsageEstimator.shallowSizeOf(futures));
```

Comment:
I think we normally do not account for any intermediated structures to avoid to go too much into details with the accounting. As we add general overhead to each accounted byte, this should already cover such intermediate objects.
Do you really see these structures as a dominant memory consumer while profiling?

Reply:
I saw it's an array and not a single instance and thought it can have an effect on big tables.

>Do you really see these structures as a dominant memory consumer while profiling?

0.04%, 2.605.792 objects in retained heap. Also, its top 3 items insides search thread (highlighed by red in the attachment)
Judging by percentage it's nothing, judging by amount and position maybe has some effect and deserved to be accounted?

<img width="779" alt="Screenshot 2024-06-12 at 15 57 43 (2)" src="https://github.com/crate/crate/assets/7446940/6b467da8-aa7a-426d-bd55-e309b7bcc0ff">

Reply:
But isn't this showing that the results of the futures consume this 0.04% ?
So beside of the shallow sizes which could maybe neglected, we may only need to account for the value of the future, such the sorted slice of the original list only?